src/commands/topics.ts

```diff
@@ -425,67 +485,34 @@ export async function produceMessage(content: any, topic: KafkaTopic): Promise<P
   const request: ProduceRecordRequest = {
     topic_name: topic.name,
     cluster_id: topic.clusterId,
-    ProduceRequest: produceRequest,
   };
 
   let response: ProduceResponse | undefined;
   let timestamp = new Date();
 
-  response = await recordsApi.produceRecord(request);
-  // we may get a misleading `status: 200` with a nested `error_code` in the response body
-  // ...but we may also get `error_code: 200` with a successful message
-  if (response.error_code >= 400) {
-    throw new ResponseError(
-      new Response("", {
-        status: response.error_code,
-        statusText: response.message,
-      }),
-    );
+  const sidecar = await getSidecar();
+
+  if (forCCloudTopic) {
+    const ccloudClient: ConfluentCloudProduceRecordsResourceApi =
+      sidecar.getConfluentCloudProduceRecordsResourceApi(topic.connectionId);
+    const ccloudResponse = await ccloudClient.gatewayV1ClustersClusterIdTopicsTopicNameRecordsPost({
+      ...request,
+      x_connection_id: topic.connectionId,
+      dry_run: false,
+      ProduceRequest: produceRequest as CCloudProduceRequest,
+    });
+    response = ccloudResponse as ProduceResponse;
+  } else {
+    // non-CCloud topic route:
+    const client: RecordsV3Api = sidecar.getRecordsV3Api(topic.clusterId, topic.connectionId);
+    response = await client.produceRecord({ ...request, ProduceRequest: produceRequest });
   }
+
   timestamp = response.timestamp ? new Date(response.timestamp) : timestamp;
 
   return { timestamp, response };
 }
 
-export async function createProduceRequestData(
-  topic: KafkaTopic,
-  key: any,
-  value: any,
-): Promise<{ keyData: ProduceRequestData; valueData: ProduceRequestData }> {
```

Comment:
Moved to https://github.com/confluentinc/vscode/blob/djs/add-schema-to-jsonschema/src/commands/utils/produceMessage.ts